viscy/data/livecell.py

```diff
@@ -87,41 +123,68 @@ def __getitem__(self, idx: int) -> Sample:
-        train_transforms: list[MapTransform] = [],
-        val_transforms: list[MapTransform] = [],
+        train_cpu_transforms: list[MapTransform] = [],
+        val_cpu_transforms: list[MapTransform] = [],
+        train_gpu_transforms: list[MapTransform] = [],
+        val_gpu_transforms: list[MapTransform] = [],
         test_transforms: list[MapTransform] = [],
         batch_size: int = 16,
         num_workers: int = 8,
+        pin_memory: bool = True,
     ) -> None:
         super().__init__()
-        self.train_val_images = Path(train_val_images)
-        if not self.train_val_images.is_dir():
-            raise NotADirectoryError(str(train_val_images))
-        self.test_images = Path(test_images)
-        if not self.test_images.is_dir():
-            raise NotADirectoryError(str(test_images))
-        self.train_annotations = Path(train_annotations)
-        if not self.train_annotations.is_file():
-            raise FileNotFoundError(str(train_annotations))
-        self.val_annotations = Path(val_annotations)
-        if not self.val_annotations.is_file():
-            raise FileNotFoundError(str(val_annotations))
-        self.test_annotations = Path(test_annotations)
-        if not self.test_annotations.is_file():
-            raise FileNotFoundError(str(test_annotations))
-        self.train_transforms = Compose(train_transforms)
-        self.val_transforms = Compose(val_transforms)
+        if train_val_images is not None:
+            self.train_val_images = Path(train_val_images)
+            if not self.train_val_images.is_dir():
+                raise NotADirectoryError(str(train_val_images))
+        if test_images is not None:
+            self.test_images = Path(test_images)
+            if not self.test_images.is_dir():
+                raise NotADirectoryError(str(test_images))
+        if train_annotations is not None:
+            self.train_annotations = Path(train_annotations)
+            if not self.train_annotations.is_file():
+                raise FileNotFoundError(str(train_annotations))
+        if val_annotations is not None:
+            self.val_annotations = Path(val_annotations)
+            if not self.val_annotations.is_file():
+                raise FileNotFoundError(str(val_annotations))
+        if test_annotations is not None:
+            self.test_annotations = Path(test_annotations)
+            if not self.test_annotations.is_file():
+                raise FileNotFoundError(str(test_annotations))
+        self._train_cpu_transforms = Compose(train_cpu_transforms)
+        self._val_cpu_transforms = Compose(val_cpu_transforms)
+        self._train_gpu_transforms = Compose(train_gpu_transforms)
+        self._val_gpu_transforms = Compose(val_gpu_transforms)
         self.test_transforms = Compose(test_transforms)
         self.batch_size = batch_size
         self.num_workers = num_workers
+        self.pin_memory = pin_memory
+
+    @property
```

Comment:
These properties are already defined on the `HCSDataModule` are they necessary here?

Reply:
This is not a subclass of `HCSDataModule`.